Modules/Networking/Core/NetworkManager.swift

```diff
@@ -62,17 +62,16 @@ extension NetworkManager {
 
 extension NetworkManager {
   /// Returns the current manifest for the chunk of TEKs exposed by the backend
-  public func getKeysIndex() -> Promise<KeysIndex> {
-    return self.request(KeysIndexRequest())
+  public func getKeysIndex(country: Country?) -> Promise<KeysIndex> {
```

Comment:
If you model `Italy` as just a specific Country, you don't need these optional. It can just be a normal Country, and inside you do 
```swift
if country.id == Country.italyID { bla bla bla } else { somethingElse }
```